Make NEP and NMEP settings user-configurable

gsketefian opened the pull request:
## DESCRIPTION OF CHANGES: 
<!-- One or more paragraphs describing the problem, solution, and required changes. -->
Currently, the settings for neighborhood vx methods (NEP and NMEP) are hard-coded in the METplus config template (jinja2) files.  This PR makes them user-configurable via the SRW's yaml configuration file.

It also adds a new WE2E test (`MET_ensemble_verification_only_vx_with_NEP_NMEP`) to test the new configurable settings.

One issue is that this PR will cause the `develop_DTC` branch diverge from the `develop` branch (which always follows the authoritative `ufs-community`'s `develop` branch) in ways that are outside the `DTC` subdirectory in `develop_DTC`.  So there will likely be conflicts in the future when we try to update the `develop` branch with that from `ufs-community`.

gsketefian committed:
Add user-configurable neighborhood probability vx parameters to the SRW App's config file.

gsketefian committed:
Modify the ex-scripts for running GenEnsProd and GridStat_ensprob to pass the user-configurable neighborhood probability (NEP and NMEP) vx parameters to the jinja2 templates for generating METplus config files.

gsketefian committed:
Modify the jinja2 templates for generating METplus config files for GenEnsProd and GridStat_ensprob such that they make use of the user-configurable neighborhood probability (NEP and NMEP) vx parameters.

gsketefian committed:
Modify the default SRW App config file such that the value of NBRHD_PROB_VALID_THRESH matches the hard-coded value being used before this user-configurable parameter was introduced.

gsketefian committed:
Minor cleanup.

gsketefian committed:
Fix typos.

gsketefian committed:
Clean up comments in config_defaults.yaml.

gsketefian committed:
Turn off NEP and NMEP calculations by default.

gsketefian committed:
Bug fix so boolean flags are not treated as strings.

gsketefian committed:
Clean up comments.

gsketefian committed:
Remove extra lines.

gsketefian committed:
Add WE2E test for calculating NEP and NMEP probabilities.

gsketefian committed:
Merge branch 'develop' into feature/configurable_nep_nmep

RatkoVasic-NOAA committed:
Update Hera with new disks (scratch{1,2} - scratch{3,4}) (#1299)

Hera is switching from /scratch1,2 disks to /scratch3,4:

* Need to update *.yaml files pointing to new directories
* Updated modulefiles/build_hera_gnu.lua to use /contrib spack-stack
* Data is already copied to new disks.

benkozi committed:
[develop] feat: AQMv8 support and update WM hash (#1294)

gsketefian committed:
Merge branch 'develop_DTC' into feature/configurable_nep_nmep

jdkublnick committed:
[develop] Updating Develop documentation with applicable changes from v3.0.0 release (#1281)

Removing MacOS/Linux/WCOSS support, updating links and other misc updates.

---------

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

michelleharrold commented on ush/config_defaults.yaml:
```diff
@@ -2746,6 +2746,130 @@ verification:
+  #
+  # NMEP_SMOOTH_GAUSSIAN_RADIUS, NMEP_SMOOTH_GAUSSIAN_DX:
+  # If performing Gaussian smoothing, these parameters are used to calculate
+  # the non-dimensional standard deviation or width, sigma, of the gaussian,
```
Should the g in Gaussian be capitalized?

gsketefian replied:
Fixed!

MichaelLueken committed:
[develop] Remove &namsfc namelist from FV3 input.nml and remove ush/set_fv3nml_sfc_climo_filenames.py from the workflow generation (#1295)

Removing the &namsfc entry from the FV3 input.nml file, as well as removing ush/set_fv3nml_sfc_climo_filenames.py from the workflow generation.

MichaelLueken committed:
[develop] Update weather model and UPP hashes, cleanup CODEOWNERS file (#1306)

* Update ufs-weather-model hash to 1f0ab43 (July 1)
* Update UPP hash to e5f9fcb (May 2)
* Cleanup .github/CODEOWNERS file and add Priya, Josh, and Brandon as code owners
* Remove Gaea C5 coverage and comprehensive WE2E suites, since the machine has been deprecated
* Remove Gaea C5, Jet, and WCOSS2 from the list of machines that tests were run on (in .github/PULL_REQUEST_TEMPLATE) since they are no longer supported
* Correct issues in the modulefiles/build_noaacloud_intel.lua file that was causing the weather model to fail to build on NOAA Cloud platforms.

jdkublnick committed:
[develop] Updating we2e script in documentation and updating Quickstart chapter.  (#1307)

* Updated WE2E script in UG
* Removed Jet and Gaea-C5 from RunSRW.rst
* Updated example from Gaea-C5 to Hercules in quckstart.rst

---------

Co-authored-by: Michael Lueken <63728921+MichaelLueken@users.noreply.github.com>
Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

drnimbusrain committed:
[develop] Updating GFSv17_p8 field table for AQMv8. (#1309)

Updating the GFSv17_p8 aqm field table for updated chemical tracers to support running these physics with AQMv8

jdkublnick committed:
[develop] Fixing dead link (#1310)

Quick link fix to resolve build errors

gsketefian committed:
Merge branch 'develop' into develop_DTC

gsketefian committed:
Fix typo.

gsketefian committed:
Merge branch 'develop_DTC' into feature/configurable_nep_nmep
